addons/nuxeo-ai-aws-core/src/main/java/org/nuxeo/ai/AWSHelper.java

```diff
@@ -0,0 +1,166 @@
+import org.nuxeo.ecm.core.storage.sql.ImageHelperWithS3;
+import org.nuxeo.runtime.api.Framework;
+import org.nuxeo.runtime.aws.NuxeoAWSCredentialsProvider;
+import org.nuxeo.runtime.aws.NuxeoAWSRegionProvider;
+import org.nuxeo.runtime.services.config.ConfigurationService;
+import com.amazonaws.AmazonServiceException;
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.services.logs.model.UnrecognizedClientException;
+import com.amazonaws.services.rekognition.model.AccessDeniedException;
+import com.amazonaws.services.rekognition.model.Image;
+
+/**
+ * Helps with S3 images and AWS credentials
+ *
+ * @since 11.1
+ */
+public class AWSHelper {
+
+
+    public static final String CONFIG_NAME = "nuxeo-ai-aws";
+
+    public static final String CONFIG_USE_S3 = "nuxeo.enrichment.aws.s3";
+
+    public static final String S3_MANAGER_NAME = "org.nuxeo.ecm.core.storage.sql.S3BinaryManager";
+
+    protected static final Set<String> FATAL_ERRORS = new HashSet<>(Arrays.asList(
+            UnrecognizedClientException.class.getSimpleName(),
+            AccessDeniedException.class.getSimpleName()));
+
+    private static final Logger log = LogManager.getLogger(AWSHelper.class);
+
+    protected static AWSHelper INSTANCE;
+
+    protected static boolean hasS3BinaryManager;
+
+    static {
+        try {
+            Class.forName(S3_MANAGER_NAME);
+            hasS3BinaryManager = true;
+        } catch (ClassNotFoundException e) {
+            hasS3BinaryManager = false;
+        }
+    }
+
+    protected final ImageHelperWithS3 s3Helper;
+
+    protected final NuxeoAWSRegionProvider regionProvider;
+
+    protected final NuxeoAWSCredentialsProvider credentialsProvider;
+
+    protected AWSHelper(ImageHelperWithS3 s3Helper, NuxeoAWSRegionProvider regionProvider,
+                        NuxeoAWSCredentialsProvider credentialsProvider) {
+        this.s3Helper = s3Helper;
+        this.regionProvider = regionProvider;
+        this.credentialsProvider = credentialsProvider;
+    }
+
+    /**
+     * Gets a reference to a single instance of the AWSHelper class.
+     */
+    public static AWSHelper getInstance() {
+        if (INSTANCE == null) {
+            NuxeoAWSRegionProvider nuxeoAWSRegionProvider = new NuxeoAWSRegionProvider(CONFIG_NAME);
+            NuxeoAWSCredentialsProvider nuxeoAWSCredentialsProvider = new NuxeoAWSCredentialsProvider(CONFIG_NAME);
+            if (hasS3BinaryManager && !Framework.getService(ConfigurationService.class)
+                                                .isBooleanPropertyFalse(CONFIG_USE_S3)) {
+                // If the binary manager is in the classpath and it hasn't been explicitly turned off, then use it.
+                INSTANCE = new AWSHelper(new ImageHelperWithS3(), nuxeoAWSRegionProvider, nuxeoAWSCredentialsProvider);
+            } else {
+                INSTANCE = new AWSHelper(null, nuxeoAWSRegionProvider, nuxeoAWSCredentialsProvider);
```

Comment:
There's a bit of duplication here, you could just set a local variable with either `new ImageHelperWithS3()` or `null` then pass that to a single instantiation of the `new AWSHelper`